### PR TITLE
Fix signing for security level 3

### PR DESCRIPTION
**Pull request by thibault-martinez**

### CODE REVIEW

**Comment by Cylix** [comment]
Can you briefly describe what is the current bug and how this change fix it?
Will help for review and for record, thanks!

**Comment by thibault-martinez** [comment]
- On https://github.com/thibault-martinez/iota.lib.cpp/pull/276/files#diff-ee63854217b7b73cff72b3241f441984L183 we were using the seed security and not the key security. 
- On https://github.com/thibault-martinez/iota.lib.cpp/pull/276/files#diff-ee63854217b7b73cff72b3241f441984L203 we weren't creating enough signature fragments in case of security level 3. It was, don't know why, limited by some * 2 (e.g https://github.com/thibault-martinez/iota.lib.cpp/pull/276/files#diff-ee63854217b7b73cff72b3241f441984L210)

File: source/crypto/signing.cpp [hunk 1]
```diff
@@ -162,64 +167,62 @@ signInputs(const Models::Seed& seed, const std::vector<Models::Address>& inputs,
...
-                Crypto::Signing::signatureFragment(secondBundleFragment, secondFragment);
-
-            //  Convert signature to trytes and assign it again to this bundle entry
-            txb.setSignatureFragments(Types::tritsToTrytes(secondSignedFragment));
-          }
+      for (unsigned int j = 1; j < keySecurity; ++j) {
+        // Because the signature is > 2187 trytes, we need to
+        // find the subsequent transaction to add the remainder of the signature
+        // Same address as well as value = 0 (as we already spent the input)
+        if (bundle[i + j].getAddress() == addr && bundle[i + j].getValue() == 0) {
```
Comment (Cylix) [inline source/crypto/signing.cpp]: Are you sure that dropping the loop and using indexing has the same effect?
That means the bundle should already be filled with empty transactions for signatures, is that the case? If so, is it order the same as what seem to be used in JS library?

**Review by Cylix — COMMENTED** [review]
Can we have an additional test for that case? If our tests passed, I guess this is not currently covered?

**Comment by Cylix** [comment]
For this one, can you confirm the assumptions made when talking about it privately / safe to merge?

**Comment by thibault-martinez** [comment]
I'll deal with this one tomorrow.

**Pull request closed without merging**